### PR TITLE
Add support for xmllint as a fixer.

### LINKED ISSUE #1831: Add support for xmllint as a fixer.
**Name:** xmllint
**URL:** http://xmlsoft.org/xmllint.html

By running --format this linter can be used as a fixer as well. As such it should be enabled as one.

### PR DESCRIPTION
**Pull request by Steap**
This fixes #1831.

### CODE REVIEW

File: autoload/ale/fixers/xmllint.vim [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+call ale#Set('xml_xmllint_indentsize', 2)
+
+function! ale#fixers#xmllint#Fix(buffer) abort
+    let l:executable = ale#Escape(ale#Var(a:buffer, 'xml_xmllint_executable'))
+    let l:filename = ale#Escape(bufname(a:buffer))
+    let l:command = l:executable . ' --format ' . l:filename
+
+    let l:indent = ale#Var(a:buffer, 'xml_xmllint_indentsize')
+    if l:indent isnot# ''
+        let l:env = 'XMLLINT_INDENT="' . repeat(' ', l:indent) . '"'
```
Comment (w0rp) [inline autoload/ale/fixers/xmllint.vim]: Use `ale#Env('XMLLINT_INDENT', repeat(' ', l:indent))` instead. The result will work on Windows or Unix systems.

File: test/fixers/test_xmllint_fixer_callback.vader [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  \     . ale#Escape(bufname(bufnr('')))
+  \ },
+  \ ale#fixers#xmllint#Fix(bufnr(''))
+
+Execute(The xmllint callback should include the XMLLINT_INDENT variable):
+  let g:ale_xml_xmllint_indentsize = 2
+
+  AssertEqual
+  \ {
+  \   'command': 'XMLLINT_INDENT="  " '
```
Comment (w0rp) [inline test/fixers/test_xmllint_fixer_callback.vader]: You can just apply `ale#Env` here again. The result will be different on Windows.

**Comment by Steap** [comment]
I updated the PR to use ale#Env as requested.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**